File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        words = s.split()  # Split the input string into words, removing extra spaces.

        reversed_words = words[::-1]  # Reverse the list of words.

        result = " ".join(reversed_words)  # Join the reversed words with a single space.

        return result

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string word;
        vector<string> words;

        for (int i = 0; i < s.length(); i++) {
            if (s[i] == ' ' && !word.empty()) {
                words.push_back(word);
                word.clear();
            } else if (s[i] != ' ') {
                word += s[i];
            }
        }

        if (!word.empty()) {
            words.push_back(word);
        }

        std::string reversedSentence;
        for (int i = words.size() - 1; i >= 0; i--) {
            reversedSentence += words[i];
            if (i > 0) {
                reversedSentence += ' ';
            }
        }

        return reversedSentence;
    }
};